### ISSUE TITLE
iFrame: Find a way to reduce screen reader verbosity

### ISSUE THREAD

**Issue by alexstine** [WordPress/gutenberg#46260]
### Description

When entering the iFrame in FSE, it can be rather verbose. The below PR adds the iFrame to the post editor, here is the difference in speech output.

#46212 

### Step-by-step reproduction instructions

Before the PR:
1. Enable NVDA.
2. Open the post editor.
3. Select a block.
4. Tab in to the sidebar.
5. Shift+tab back to the block.

Speech sample:
> Editor content  region
  Add title  edit  multi line  Testing placeholder
  Paragraph block  document  editable  This is some content.
  Editor settings  region    list  with 2 items  Post  button  
  Editor content  region
  Paragraph block  document  editable  This is some content.

After the PR:
1. Enable NVDA.
2. Open the post editor.
3. Select a block.
4. Tab in to the sidebar.
5. Shift+tab back to the block.

Speech sample:
> Editor content  region
  Editor canvas  frame
  Editor canvas  document
  Add title  edit  multi line  Testing placeholder
  Paragraph block  document  editable  Testing content.
  Editor settings  region    list  with 2 items  Post  button  
  Editor content  region
  Editor canvas  frame
  Editor canvas  document
  Paragraph block  document  editable  Testing content.

There is now two lines more of verbosity than before and it is not even information that helps the user. Our job is to keep things flowing, not bog users down with useless information.

### Environment info

Gutenberg: trunk
Browser: Firefox and Google Chrome
Screen reader: NVDA (default verbosity)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by ellatrix** [comment]
@alexstine Do you have any ideas on how we could do this? We could leave the iframe title empty (instead of `Editor canvas`) if it's useless, not sure if that would be good or bad.

**Comment by alexstine** [comment]
@ellatrix Not sure. That could be worth a try. I also thought this might be a fun project to work on. What would you think about attempting to wrap the iFrame in `role="application"` like the classic editor? Maybe then the editor will be easier to use if screen readers are forced to send keyboard events vs. mode switching. It will be a lot of work for us to manage the experience but better late than never. Slack has seemed to have some success in wrapping only certain areas of their web client/website in application roles, I think it is time we give it a try if you are onboard for the work. I am hoping in January I will have additional dev capacity.

Thanks.

**Comment by joedolson** [comment]
Has anything been done to work on this yet? It seems like at least testing the empty iframe title is worth trying, although I believe that actually makes announcements worse on NVDA, unless you disable iframe announcements.

**Comment by alexstine** [comment]
@joedolson I think we might try adding `role="application"` to the iFrame? This will help us mirror what currently happens in classic editor. What really worries me about this approach though, this could really break what little keyboard support there is for some blocks. However, it will allow us much more control over verbosity, focus, labeling, etc.

**Comment by ellatrix** [comment]
> What would you think about attempting to wrap the iFrame in role="application" like the classic editor? Maybe then the editor will be easier to use if screen readers are forced to send keyboard events vs. mode switching. It will be a lot of work for us to manage the experience but better late than never. Slack has seemed to have some success in wrapping only certain areas of their web client/website in application roles, I think it is time we give it a try if you are onboard for the work. I am hoping in January I will have additional dev capacity.

Sorry, missed this message in all the notifications. I'm up for that! I'm not sure what it would involve but I'm happy to help!

**Comment by alexstine** [comment]
@ellatrix We'll need to start by adding `role="application"` to the iFrame and then testing everything a lot. Won't know how badly this can break things until we get over that first step. Essentially, need to make the editor canvas act more like TinyMCE for screen readers.

**Comment by ellatrix** [comment]
Here's the first step then :)

* https://github.com/WordPress/gutenberg/pull/53364
